File: 프로그래머스/lv2/12913. 땅따먹기/땅따먹기.java
import java.util.*;

class Solution {
    int solution(int[][] board) {
        int size = board.length;
        for(int i=1; i<size; i++) {
            board[i][0] += Math.max(board[i-1][1], Math.max(board[i-1][2], board[i-1][3]));
            board[i][1] += Math.max(board[i-1][0], Math.max(board[i-1][2], board[i-1][3]));
            board[i][2] += Math.max(board[i-1][0], Math.max(board[i-1][1], board[i-1][3]));
            board[i][3] += Math.max(board[i-1][0], Math.max(board[i-1][1], board[i-1][2]));
        }
        return Math.max(board[size-1][0], Math.max(board[size-1][1], Math.max(board[size-1][2], board[size-1][3])));
    }
}

File: 프로그래머스/lv2/12913. 땅따먹기/땅따먹기.py
def solution(land):
    answer = 0
    n = len(land)
    # 1. 그냥... 이전 합으로 비교하면서 풀기? 시초각인데
    for i in range(1, n):
        for j in range(4):
            land[i][j] = land[i][j] + max(land[i-1][:j] + land[i-1][j+1:])
    answer = max(land[n-1])

    return answer